### DISCUSSION TITLE
Ethereum Merkel Proof

### DISCUSSION THREAD

**Discussion by VahanN95** [smartcontractkit/full-blockchain-solidity-course-py#1208] [Q&A] [answered]
Hi Everyone, 

I was reading about Merkel Proof and how the lightweight node can proof if the transaction exists in the block. 
So the whole algorithm is clear but I have some confusion.

1. The question is, what exactly the lightweight node requesting from the full node?
2. How the full node is providing the Merkel proof? 

So lets say we have  A,B,C,D transactions in the block and the Merkel root is HashABCD and lightweight no wants to validate if transaction D exists in the block. So how the communication is done and how the full node understands from which leaf node it should generate the merkel proof?

**Comment by cromewar** [comment]
@PatrickAlphaC a  hand on this please.

**Comment by PatrickAlphaC** [comment, marked as answer]
Hi @VahanN95! Could you actually ask this on stack exchange ETH? Love the question, but it's a bit out of the scope of this course.

Thanks!

https://ethereum.stackexchange.com/
